components/command/filemanager.go

```diff
@@ -69,8 +69,34 @@ func (fm *FileManager) CopyFile(localPath, remotePath string, opts ...pulumi.Res
 	return fm.runner.NewCopyFile(localPath, remotePath, opts...)
 }
 
-func (fm *FileManager) CopyInlineFile(fileContent pulumi.StringInput, remotePath string, useSudo bool, opts ...pulumi.ResourceOption) (*remote.Command, error) {
-	return fm.command.CopyInlineFile(fm.runner, fileContent, remotePath, useSudo, opts...)
+func (fm *FileManager) CopyInlineFile(fileContent pulumi.StringInput, remotePath string, opts ...pulumi.ResourceOption) (*remote.CopyFile, error) {
+	var waitingGroup sync.WaitGroup
+	waitingGroup.Add(1)
+
+	var tempFilePath string
+
+	// Write the content into a temporary file and get the path
+	fileContent.ToStringOutput().ApplyT(func(content string) error {
+		defer waitingGroup.Done()
+
+		tempFile, err := os.CreateTemp("", filepath.Base(remotePath))
+		if err != nil {
+			return err
+		}
+		defer tempFile.Close()
+
+		tempFilePath = tempFile.Name()
+		_, err = tempFile.WriteString(content)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	waitingGroup.Wait()
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
I suspect this blocks in preview mode. The proper way would be to return the count of written bytes from `tempFile.WriteString(content)` as a `IntOutput`, which you can depend on when calling `fm.CopyFile`

Reply:
Yes this blocks to be sure the file is properly written when we declare the `CopyFile` right after, otherwise we could get a "file not found" error. 

I don't know how to depend on an `IntOutput`, I'll have a look. Looks like I can't just `utils.MergeOptions(opts, pulumi.DependsOn([]pulumi.Resource{output}))`

Reply:
I broke the CI so yeah there is definitely something wrong